File: Exemplos/27-prjExemplosGuiObjetos/src/fatec/poo/model/Retangulo.java
package fatec.poo.model;

/**
 *
 * @author ltshi
 */

public class Retangulo {
    private double altura;
    private double base;
    private String unidadeMedida;
    
    public Retangulo (String uniMed){
        unidadeMedida = uniMed;
    }
    
    public void setAltura(double a) {
        altura = a;
    }
    
    public void setBase(double b) {
        base = b;
    }
    
    public double getAltura() {
        return altura;
    }
    
    public double getBase() {
        return base;
    }
    
    public String getUniMed() {
        return unidadeMedida;
    }
    
    public double calcArea() {
        return(altura*base);
    }
    
    public double calcPerimetro() {
        return((altura+base)*2);
    }
    
    public double calcDiagonal() {
        return (Math.sqrt(Math.pow(base, 2) + Math.pow(altura, 2)));
    }
}

File: Exemplos/prjAvalPOO_AD_G1_FilipeAffonso/prjExemploConstrutor/Retangulo.cpp
#include "Retangulo.h"
#include <math.h>
                 
Retangulo::Retangulo(double b, double a)
{
  Base=b;
  Altura=a;                          
}          
                 
double Retangulo::getBase()
{
  return(Base);     
}              
double Retangulo::getAltura() 
{
  return(Altura);
}  
double Retangulo::calcArea()
{
  double Area;
  Area = Base * Altura;
  return(Area);     
} 
double Retangulo::calcPerimetro()
{
  double Perimetro;
  Perimetro = 2 * (Base + Altura); 
  return(Perimetro);    
}             

double Retangulo::calcDiagonal()
{
   double Diagonal;
   Diagonal = sqrt((Base*Base) + (Altura*Altura));    
   return(Diagonal);
}




